synapse/rest/admin/background_updates.py

```diff
@@ -105,3 +110,68 @@ async def on_GET(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
             }
 
         return 200, {"enabled": enabled, "current_updates": current_updates}
+
+
+class BackgroundUpdateStartJobRestServlet(RestServlet):
+    """Allows to start specific background updates"""
+
+    PATTERNS = admin_patterns("/background_updates/start_job")
+
+    def __init__(self, hs: "HomeServer"):
+        self._is_mine_id = hs.is_mine_id
+        self._auth = hs.get_auth()
+
+        self._store = hs.get_datastore()
+
+    async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
+        requester = await self._auth.get_user_by_req(request)
+        await assert_user_is_admin(self._auth, requester.user)
+
+        body = parse_json_object_from_request(request)
+        assert_params_in_dict(body, ["job_name"])
+
+        job_name = body.get("job_name")
```

Comment:
We know the `job_name` key must exist, so best to just get it.

```suggestion
        job_name = body["job_name"]
```